cupy/cuda/function.pyx

```diff
@@ -18,6 +19,7 @@ from cupy_backends.cuda.api cimport runtime
 from cupy.cuda cimport stream as stream_module
 from cupy.cuda.memory cimport MemoryPointer
 from cupy.cuda.texture cimport TextureObject, SurfaceObject
+from cupy.cuda import device
```

Comment:
Question: can this be cimport'd? I guess not, because `Device.attributes` is a Python property, but I'd like to check 🙂

Reply:
I tried cimport, but an error occurs at build time.
```
Error compiling Cython file:
------------------------------------------------------------
...
    cdef int dev_id
    cdef int num_sm
    cdef int max_grid_size
    if enable_cooperative_groups:
        dev_id = device.get_device_id()
        num_sm = device._get_attributes(dev_id)['MultiProcessorCount']
                      ^
------------------------------------------------------------

cupy/cuda/function.pyx:166:23: cimported module has no attribute '_get_attributes'
```

Reply:
Yeah it's because `_get_attributes` is not cdef'd. I guess it's ok to stay as is, unless we later find out the overhead is not negligible, in which case we can always change to the runtime API.